subsys/logging/log_core.c

```diff
@@ -24,9 +24,14 @@ LOG_BACKEND_UART_DEFINE(log_backend_uart);
 
 static struct log_list_t list;
 static bool panic_mode;
-static bool initialized;
 
-static timestamp_get_t timestamp_func;
+static u32_t dummy_timestamp(void);
+static timestamp_get_t timestamp_func = dummy_timestamp;
+
+static u32_t dummy_timestamp(void)
```

Comment:
Is timestamp now mandatory? If not, why we have to have default function?

Reply:
it is mandatory. and actually there are 2 default functions. Dummy_timestamp is used between log_core_init() and log_init(). Then timestamp which is using k_cycle_get_32 is used and it can be later on replaced by a user function. k_cycle_get_32 cannot be used from the start because it may not be configured/available then.